src/builder.ts

```diff
@@ -102,16 +121,20 @@ export default class Builder {
 
 			// Catch any errors the stream produces
 			outputStream.on('error', (err: Error) => {
+				errored = true
 				self.callHook(hooks, 'buildFailure', handler, err, self.layers)
 			})
 			dup.on('error', (err: Error) => {
```

Comment:
This handler is identical between `outputStream` and `dup`, if you could extract it to a shared function that'd be great

Reply:
Thanks @Page- , but a few days ago I had a different commit with several such arguable improvements including removing the dependency on `event-stream`, but in a devexp checkin with @CameronDiver we decided it was better to minimise the  changes so I reverted it all to the minimum necessary to fix the bug. At this point I just want to get this PR out of the way for the build secrets feature as customers and support agents and Shaun and Hedley with the NUCs have all been waiting and asking for it... :-)